pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -280,10 +281,52 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 	return nil
 }
 
+func findMachineAddress(ctx context.Context, in clusterapi.PostProvisionInput, key crclient.ObjectKey) (string, error) {
+	powerVSMachine := &capibm.IBMPowerVSMachine{}
+
+	// Get the machine address
+	// Unfortunately https://github.com/gomodules/wait/blob/v0.2.0/wait.go#L205
+	// can only return a bool.  It would be nice if it could return a pointer.
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: time.Second * 10,
+		Factor:   float64(1.5),
+		Steps:    32,
+	}, func(ctx context.Context) (bool, error) {
+		if err := in.Client.Get(ctx, key, powerVSMachine); err != nil {
+			return false, fmt.Errorf("failed to get PowerVS machine in PostProvision: %w", err)
+		}
+
+		for _, address := range powerVSMachine.Status.Addresses {
+			if address.Type == corev1.NodeInternalIP {
+				logrus.Debugf("PostProvision: found %s address %s", key.Name, address.Address)
+				return true, nil
+			}
+		}
+
+		logrus.Debugf("PostProvision: waiting for %s machine", key.Name)
+		return false, nil
+	}); err != nil {
+		return "", err
+	}
+
+	if err := in.Client.Get(ctx, key, powerVSMachine); err != nil {
+		return "", fmt.Errorf("failed to get PowerVS machine in PostProvision: %w", err)
+	}
+
+	for _, address := range powerVSMachine.Status.Addresses {
+		if address.Type == corev1.NodeInternalIP {
+			return address.Address, nil
+		}
+	}
+
+	return "", fmt.Errorf("failed to get machine %s IP address", key.Name)
```

Comment:
Not better, I just don't see the need for the complexity of an exponential backoff instead of a simple interval polling given that we're just checking for a condition in the local controlplane. But it's up to you which one to use.

Reply:
I made the switch.